feat(notification): реализовать WsJwtGuard для авторизации WebSocket

i-bocharov opened the pull request:
- Создан guard для проверки JWT-токена из query-параметра
- Добавлены типы и интерфейсы WsClient, JwtPayload, NotificationType, NotificationPayload
- Интеграция с UsersService.findUserById
- Обработка ошибок: отсутствие токена, невалидный токен, пользователь не найден

i-bocharov committed:
feat(notification): реализовать WsJwtGuard для авторизации WebSocket

- Создан guard для проверки JWT-токена из query-параметра
- Добавлены типы и интерфейсы WsClient, JwtPayload, NotificationType, NotificationPayload
- Интеграция с UsersService.findUserById
- Обработка ошибок: отсутствие токена, невалидный токен, пользователь не найден

RolandSallaz commented on src/notification/guards/ws-jwt.guard.ts:
```diff
@@ -0,0 +1,44 @@
+    const token = client.handshake.query?.token;
+
+    if (!token) {
+      throw new UnauthorizedException('Token is missing');
```
Сокет не понимает HTTP ошибки, поэтому выкидывать нужно WsException, вот пример `new WsException('Token is missing')`

RolandSallaz commented on src/notification/guards/ws-types.ts:
```diff
@@ -0,0 +1,41 @@
+/**
+ * Payload для JWT токена
+ */
+export interface JwtPayload {
```
Можно переиспользовать JwtPayload из auth.types, он вполне подойдет

RolandSallaz commented on src/notification/guards/ws-jwt.guard.ts:
```diff
@@ -0,0 +1,44 @@
+
+    try {
+      const payload = this.jwtService.verify<JwtPayload>(token, {
+        secret: this.configService.get('JWT_SECRET'),
```
Конфиг стоит импортнуть по ключу с типизацией

```ts
    @Inject(конфиг.KEY)
    private readonly config: ТипКонфига,
```

i-bocharov committed:
fix(notification): заменить UnauthorizedException на WsException, использовать jwtConfig для секрета токена

- WebSocket не поддерживает HTTP-статусы
- Все ошибки авторизации теперь выбрасываются как WsException
- Добавлен инжект jwtConfig через @Inject(jwtConfig.KEY)
- Используем jwtConfig.accessToken как секрет для verify()
- Удалена зависимость от ConfigService
- Удалён дублирующийся JwtPayload из ws-types.ts
- Импортирован TJwtPayload из src/auth/types
